python-package/lightgbm/basic.py

```diff
@@ -1308,6 +1315,11 @@ def __setstate__(self, state):
             state['handle'] = handle
         self.__dict__.update(state)
 
+    def _reverse_update_params(self):
```

Comment:
put this into booster is a little bit confusing. you can change the name to _reverse_update_params_of_dataset or something else. Or remove this function.